packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -31,12 +31,62 @@ import { InternalHooksManager } from '../InternalHooksManager';
 import { externalHooks } from '../Server';
 import { getLogger } from '../Logger';
 import type { WorkflowRequest } from '../requests';
-import { isBelowOnboardingThreshold } from '../WorkflowHelpers';
+import { getSharedWorkflowIds, isBelowOnboardingThreshold } from '../WorkflowHelpers';
 import { EEWorkflowController } from './workflows.controller.ee';
+import { validate as jsonSchemaValidate } from 'jsonschema';
 
 const activeWorkflowRunner = ActiveWorkflowRunner.getInstance();
 export const workflowsController = express.Router();
 
+const schemaGetWorkflowsQueryFilter = {
+	$id: '/IGetWorkflowsQueryFilter',
+	type: 'object',
+	properties: {
+		id: { anyOf: [{ type: 'integer' }, { type: 'string' }] },
+		name: { type: 'string' },
+		active: { type: 'boolean' },
+		nodes: { type: 'string' },
+		connections: { type: 'string' },
+		createdAt: {
+			type: { type: 'string' },
+			format: 'date-time',
+		},
+		updatedAt: {
+			type: { type: 'string' },
+			format: 'date-time',
+		},
+		settings: { type: 'string' },
+		staticData: { type: 'string' },
+		pinData: { anyOf: [{ type: 'integer' }, { type: 'string' }] },
+	},
+};
+
+const allowedWorkflowsQueryFilterFields = [
+	'id',
+	'name',
+	'active',
+	'nodes',
+	'connections',
+	'createdAt',
+	'updatedAt',
+	'settings',
+	'staticData',
+	'pinData',
+];
+
+interface IGetWorkflowsQueryFilter {
+	id?: number | string;
+	name?: string;
+	active?: boolean;
+	nodes?: string;
+	connections?: string;
+	createdAt?: string;
+	updatedAt?: string;
+	settings?: string;
+	staticData?: string;
+	pinData?: string;
```

Comment:
Currently we don't use any columns for filtering, but I'd suggest leaving only simple ones such as `name` and `active`. Filtering by any other column is very cumbersome right now.

Reply:
left only id, name, active